examples/extended_validation/src/tests.rs

```diff
@@ -0,0 +1,89 @@
+use rocket::testing::MockRequest;
+use rocket::http::Method::*;
+use rocket::http::{ContentType, Status};
+
+use super::rocket;
+
+fn test_login(username: &str,
+              password: &str,
+              age: &str,
+              status: Status,
+              body: Option<&'static str>) {
+    let rocket = rocket();
+    let query = &format!("username={}&password={}&age={}", username, password, age);
+
+    let mut req = MockRequest::new(Post, "/login")
+        .header(ContentType::Form)
+        .body(&query);
+
+    let mut response = req.dispatch_with(&rocket);
+    let body_str = response.body().and_then(|body| body.into_string());
+
+    println!("Checking: {:?}/{:?}/{:?}/{:?}", username, password, age, body_str);
+    assert_eq!(response.status(), status);
+
+    if let Some(string) = body {
+        assert!(body_str.map_or(true, |s| s.contains(string)));
```

Comment:
We can remove the `body_str` declaration a few lines above and write this as something like:

```rust
assert_eq!(response.body().map(|body| body.contains(string)), Some(true))
```